### PR TITLE
Remove Dokka generation content from KDoc and Dokka page

### PR DESCRIPTION
**Pull request by sarahhaggarty**
As we introduce Dokka's documentation, this PR is to update the KDoc and Dokka page to remove any duplicating content and focus scope on KDoc.

### CODE REVIEW

File: docs/topics/kotlin-doc.md
Comment (IgnatBeresnev) [inline docs/topics/kotlin-doc.md:144]: Oof, I don't think this part is actually covered in Dokka's documentation, it might instead link to this section, actually. And I'm not 100% sure it should migrate there

The primary concern is that this format might already be used by other tools and thus relied upon. I'm afraid we'd be rug pulling if Dokka became its owner and started changing things for its own use.

The safest option would be to re-write this section to be tool-independent, i.e just describe what it could be used for and what the format is, and then mention at the end that Dokka can understand it. 

But since there's no predefined name/path for such files (unlike with Java's `package-info.java`), I don't know how useful it'll be and whether we should keep it here

I'll ask for a second opinion and come back to you

File: docs/topics/kotlin-doc.md
Comment (IgnatBeresnev) [inline docs/topics/kotlin-doc.md:5]: Just thinking out loud: I wonder if we're past the point of needing to compare Kotlin with Java. It totally made sense in the beginning as people were primarily migrating from Java, but how many users/beginners are there now that are not familiar with Java and Javadoc?

**Comment by IgnatBeresnev** [comment]
We discussed that this could be a good chance to revisit the whole page and see if it needs any additional changes. My thoughts and observations, do tell me if I need to create separate tickets for any of them:

___

I can't find the issue now, but a user came to Dokka's issues asking about the `@property` tag. They were just starting out with Kotlin and presumably never coded in Java.

The user wanted to annotate class properties, they saw [the example](https://kotlinlang.org/docs/kotlin-doc.html#kdoc-syntax) and did this:

```kotlin
/**
 * @property myProperty description
 * for my property
 */
class MyClass {

    val myProperty = "value"
}
```

Whereas it's not against the spec, the more common approach would be

```kotlin
class MyClass {

    /**
     * description
     * for my property
     */
    val myProperty = "value"
}
```

^ this is obvious and comes natural to those with some Java background as it's the only way to annotate fields, but it's not clear for beginners without Java experience.

When I told them the same thing as I outlined above, the user said they had documented a lot of code this way already and probably wouldn't change it.

I'd add an example of documenting standalone properties to the example, as well as examples for some other popular tags.

___

Currently, none of the tags have examples, and for beginners it might not be obvious how they can be used, for instance see https://github.com/Kotlin/dokka/issues/2595

I'd be happy to help come up with small and universal examples
___

> @see identifier﻿
> Adds a link to the specified **class or method** to the See also block of the documentation.

I don't think it's limited to classes or methods only. It works with properties, enum entries, interfaces, typealiases. IntelliJ IDEA also resolves packages and parameters. Should it be changed to `element` to avoid possible confusion?

File: docs/topics/kotlin-doc.md
Comment (IgnatBeresnev) [reply docs/topics/kotlin-doc.md:144]: We've discussed it with the team and came to the conclusion that it's unlikely that anyone other than Dokka is using it, and it's unlikely that this will be part of the KDoc specification itself in its current state.

So we've agreed to move it to Dokka's documentation. I can't think of a place where we can put it though, since it's runner-independent, so it looks like it's going to be a separate page altogether

@sarahhaggarty do you have any objections with this? I can move it by the time we deploy documentation live, but you can do it yourself if you have the time

Comment (sarahhaggarty) [reply docs/topics/kotlin-doc.md:144]: @IgnatBeresnev, no objections from me. Do you know if we have an example of this I could preview somewhere?

File: docs/topics/kotlin-doc.md
Comment (sarahhaggarty) [reply docs/topics/kotlin-doc.md:5]: I actually asked this question recently and apparently the majority of users are still Java developers migrating to Kotlin. So I think for now we should keep it as it is. Perhaps if the audience type shifts in future we can reconsider.

**Comment by sarahhaggarty** [comment]
> We discussed that this could be a good chance to revisit the whole page and see if it needs any additional changes. My thoughts and observations, do tell me if I need to create separate tickets for any of them:

@IgnatBeresnev, thanks for taking the time to complete such a thorough review of the page! I've captured your feedback in [KT-55998](https://youtrack.jetbrains.com/issue/KT-55998/Update-KDoc-page). There are some other KDoc related tickets that I'd like to combine with this to revamp the page. I'll do the page revamp in a separate PR as it will need a lot of review from different people. I will definitely ask you about some good examples to add as part of that 😄

File: docs/topics/kotlin-doc.md
Comment (IgnatBeresnev) [reply docs/topics/kotlin-doc.md:5]: Thanks for sharing!

**Review by IgnatBeresnev — APPROVED** [review]
Looking good!

**Comment by IgnatBeresnev** [comment]
> There are some other KDoc related tickets that I'd like to combine with this to revamp the page

That's awesome to hear, thanks! I see there are some linked issues with changes that I wanted to propose, but hesitated (like adding examples for code blocks)

**Review by danil-pavlov — APPROVED** [review]

**Pull request merged**
